core/invoker/src/main/scala/whisk/core/containerpool/docker/DockerContainer.scala

```diff
@@ -142,6 +142,44 @@ class DockerContainer(protected val id: ContainerId, protected val addr: Contain
     docker.rm(id)
   }
 
+  private def isOomKilled(retries: Int = (waitForOomState / filePollInterval).toInt)(
+    implicit transid: TransactionId): Future[Boolean] = {
+    docker.isOomKilled(id)(TransactionId.invoker).flatMap { killed =>
+      if (killed) Future.successful(killed)
+      else if (retries > 0) akka.pattern.after(filePollInterval, as.scheduler)(isOomKilled(retries - 1))
+      else Future.successful(killed)
+    }
+  }
+
+  override protected def callContainer(path: String, body: JsObject, timeout: FiniteDuration, retry: Boolean = false)(
+    implicit transid: TransactionId): Future[RunResult] = {
+    val started = Instant.now()
+    val http = httpConnection.getOrElse {
+      val conn = new HttpUtils(s"${addr.host}:${addr.port}", timeout, 1.MB)
+      httpConnection = Some(conn)
+      conn
+    }
+    Future {
+      http.post(path, body, retry)
+    }.flatMap { response =>
+      val finished = Instant.now()
+
+      response.left
+        .map {
+          case t: ConnectionError =>
+            isOomKilled().map {
+              case true =>
+                logging.info(this, "connection abruptly terminated, checking for memory exhaustion")
```

Comment:
Hmm. Log message say checking for oom but here already know it is.

Reply:
True, I removed the message as our existing logging catches the situation anyway:

```
[2017-10-04T06:13:16.909Z] [INFO] [#tid_63] [DockerContainer] running result: MemoryExhausted() [marker:invoker_activationRun_finish:780:110]
```